src/software/jetson_nano/services/motor.h

```diff
@@ -169,39 +204,100 @@ class MotorService
+    struct MotorFaultIndicator checkDriverFault(uint8_t motor);
+
+    /**
+     * Spin each motor of the NUM_DRIVE_MOTORS in open loop mode to check if the encoder
+     * is responding as expected. Allows us to do a basic test of whether the encoder is
+     * physically connected to the motor board.
+     *
+     * Leaves the motors connected in MOTION_MODE_VELOCITY
      */
-    bool checkDriverFault(uint8_t motor);
+    void checkEncoderConnections();
+
+    /**
+     * Return a MotorStatus proto filled with motor faults. Some values of the proto are
+     * previously cached velocities due to SPI transaction costs.
+     *
+     * @param front_left_wheel_velocity_mps     the front left motor's velocity in
+     * mechanical MPS
+     * @param front_right_velocity_mps          the front right motor's velocity in
+     * mechanical MPS
+     * @param back_left velocity_mps            the back left motor's velocity in
+     * mechanical MPS
+     * @param back_right_velocity_mps           the back right motor's velocity in
+     * mechanical MPS
+     * @param dribbler_rpm                      the dribbler's rotations per minute
+     *
+     * @return a MotorStatus proto with the velocity of each motor as well as their fault
+     * statuses (some faults may be cached)
+     */
+    TbotsProto::MotorStatus updateMotorStatus(double front_left_velocity_mps,
+                                              double front_right_velocity_mps,
+                                              double back_left_velocity_mps,
+                                              double back_right_velocity_mps,
+                                              double dribbler_rpm);
+
+    // to check if the motors have been calibrated
+    bool is_initialized_ = false;
 
     // Select between driver and controller gpio
-    Gpio spi_demux_select_0;
-    Gpio spi_demux_select_1;
+    Gpio spi_demux_select_0_;
+    Gpio spi_demux_select_1_;
 
     // Enable driver gpio
-    Gpio driver_control_enable_gpio;
-    Gpio reset_gpio;
+    Gpio driver_control_enable_gpio_;
+    Gpio reset_gpio_;
 
     // Transfer Buffers
-    uint8_t tx[5] = {0};
-    uint8_t rx[5] = {0};
+    uint8_t tx_[5] = {0};
+    uint8_t rx_[5] = {0};
 
     // Transfer State
-    bool transfer_started  = false;
-    bool currently_writing = false;
-    bool currently_reading = false;
-    uint8_t position       = 0;
-
-    // Constants
-    RobotConstants_t robot_constants_;
+    bool transfer_started_  = false;
+    bool currently_writing_ = false;
+    bool currently_reading_ = false;
+    uint8_t position_       = 0;
 
     // SPI File Descriptors
-    std::unordered_map<int, int> file_descriptors;
+    std::unordered_map<int, int> file_descriptors_;
 
     // Drive Motors
-    EuclideanToWheel euclidean_to_four_wheel;
+    EuclideanToWheel euclidean_to_four_wheel_;
     std::unordered_map<int, bool> encoder_calibrated_;
+    std::unordered_map<int, MotorFaultIndicator> cached_motor_faults_;
```

Comment:
nima: nit: it seems like all the fields except a few don't have the trailing underscore. Lets make that consistent throughout

Reply:
made all class members with trailing underscore @nimazareian